.github/workflows/compatibility.yml

```diff
@@ -0,0 +1,85 @@
+name: Integration with Compatibility
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
+
+jobs:
+  test-opensearch:
+    env:
+      TEST_OPENSEARCH_SERVER: http://localhost:9250
+      PORT: 9250
+    strategy:
+      fail-fast: false
+      matrix:
+        cluster-version: [ "1.0.0", "1.0.1", "1.1.0", "1.2.0", "1.2.1", "1.2.2", "1.2.3", "1.2.4", "1.3.0", "1.3.1" ]
+    runs-on: ubuntu-latest
```

Comment:
Are we expecting clients to work on multiple platforms, cpu architectures? 
I am curious how this works for clients. Are there any platform specific code in clients?

As a developer I would like to have these clients tested and validated in popular platforms.

Reply:
I haven't yet come across platform specific code in the clients, we can definitely create an issue in `opensearch-clients` to add this testing as a feature.

Reply:
i believe it is interpreted at run time. But i like the idea of including strategy to cover mac/linux/windows to be proactive in all our clients.

Reply:
Sure, I'll open an issue.